packages/playwright-test/src/reporters/base.ts

```diff
@@ -258,8 +258,8 @@ export function formatFailure(config: FullConfig, test: TestCase, options: {inde
             resultLines.push('');
           }
         } else {
-          if (attachment.contentType.startsWith('text/')) {
-            let text = attachment.body!.toString();
+          if (attachment.contentType.startsWith('text/') && attachment.body) {
```

Comment:
Let me know if fixing this in base reporter is too tactical. The alternative I considered was filtering out any attachments where the `body XOR path` precondition failed to be met which would happen here https://github.com/microsoft/playwright/blob/1f6b84f445597655b015df52cef0784d9963492c/packages/playwright-test/src/workerRunner.ts#L415 and then guarantee any reporters that deal with attachments can always depend on their either being a defined path or body (like the new `async testInfo.attach` guarantees).

Reply:
Since our attachment type has `{  path?: string, body?: Buffer }`, I think that reporters should handle the case where both are missing.